src/emc/usr_intf/stepconf/build_HAL.py

```diff
@@ -23,393 +23,423 @@
 import os
 import time
 import shutil
+from stepconf.definitions import *
 
 class HAL:
     def __init__(self,app):
-        # access to:
-        self.d = app.d  # collected data
-        global SIG
-        SIG = app._p    # private data (signal names)
-        self.a = app    # The parent, stepconf
+		# access to:
+		self.d = app.d  # collected data
+		self.a = app    # The parent, stepconf
 
     def write_halfile(self, base):
-        inputs = self.a.build_input_set()
```

Comment:
because this code was reindented, the code is unreviewable since git shows the change as "remove all old lines, add all new lines".  In the future, if changes to indentation level are needed, reindenting the code (without any functional changes) should be a separate pull request so that it is possible to review the substantive changes.  However, it is probably not possible to fix this for this PR, so we'll have to live with it.